src/components/Comment/index.spec.tsx

```diff
@@ -55,4 +57,64 @@ describe(__filename, () => {
       comment.content,
     );
   });
+
+  describe('keydown protection', () => {
+    const createFakeTextareaRef = () => {
+      return {
+        ...React.createRef(),
+        current: {
+          ...document.createElement('textarea'),
+          addEventListener: jest.fn(),
+          removeEventListener: jest.fn(),
+        },
+      };
+    };
+
+    it('adds keydown listeners on mount', () => {
+      const fakeRef = createFakeTextareaRef();
+      const root = render({ createTextareaRef: () => fakeRef });
+
+      expect(fakeRef.current.addEventListener).toHaveBeenCalledWith(
+        'keydown',
+        (root.instance() as Comment).keydownListener,
+      );
+    });
+
+    it('stops propagating keydown events', () => {
+      const fakeRef = createFakeTextareaRef();
+      render({ createTextareaRef: () => fakeRef });
+
+      expect(fakeRef.current.addEventListener).toHaveBeenCalled();
+      const listenToKeydown = fakeRef.current.addEventListener.mock.calls[0][1];
+
+      const fakeEvent = createFakeEvent();
+      listenToKeydown(fakeEvent);
+
+      expect(fakeEvent.stopPropagation).toHaveBeenCalled();
+    });
+
+    it('removes keydown event listeners on unmount', () => {
+      const fakeRef = createFakeTextareaRef();
+      const root = render({ createTextareaRef: () => fakeRef });
+      const { keydownListener } = root.instance() as Comment;
+      root.unmount();
+
+      expect(fakeRef.current.removeEventListener).toHaveBeenCalledWith(
+        'keydown',
+        keydownListener,
+      );
+    });
+
+    it('does not add/remove listeners with an undefined ref', () => {
+      // Make sure this doesn't throw an error.
+      const root = render({ createTextareaRef: () => undefined });
+      root.unmount();
+    });
+
+    it('does not add/remove listeners with a null DOM node', () => {
+      // Make sure this doesn't throw an error.
+      const root = render({ createTextareaRef: () => React.createRef() });
```

Comment:
I'm a bit confused by this test. This seems to be doing exactly the same thing as the default behaviour when `createTextareaRef` is undefined, i.e., it returns `React.createRef()` (which is the same as what's done on line 21 of `Comment/index.tsx`), so why is this equivalent to "with a null DOM node"?

Reply:
It's different because this test will define a `this.textareaRef` object with `current` set to `null` (i.e. a `null` DOM node). It will exercise the second part of this if statement:

https://github.com/mozilla/addons-code-manager/blob/63ef0cffa1bf0b1a06c3430db6d6576548424539/src/components/Comment/index.tsx#L39

If you meant "this is the same as the default rendering state" then that's true but it could be risky to rely on other tests to implicitly cover this behavior (they *probably* will but they might not). I wanted to add an explicit test for it so that any breakage might be more obvious.